server/zio-http-server/src/main/scala/sttp/tapir/server/ziohttp/ZioHttpInterpreter.scala

```diff
@@ -31,61 +31,59 @@ trait ZioHttpInterpreter[R] {
 
-    def handleRequest(req: Request, filteredEndpoints: List[ZServerEndpoint[R & R2, ZioStreams with WebSockets]]) =
-      Handler.fromZIO {
-        val interpreter = new ServerInterpreter[ZioStreams with WebSockets, RIO[R & R2, *], ZioResponseBody, ZioStreams](
-          _ => filteredEndpoints,
-          zioHttpRequestBody,
-          zioHttpResponseBody,
-          interceptors,
-          zioHttpServerOptions.deleteFile
-        )
+    def handleRequest(req: Request, filteredEndpoints: List[ZServerEndpoint[R & R2, ZioStreams with WebSockets]]): ZIO[R & R2, Response, Response] = {
+      val interpreter = new ServerInterpreter[ZioStreams with WebSockets, RIO[R & R2, *], ZioResponseBody, ZioStreams](
+        _ => filteredEndpoints,
+        zioHttpRequestBody,
+        zioHttpResponseBody,
+        interceptors,
+        zioHttpServerOptions.deleteFile
+      )
 
-        interpreter
-          .apply(ZioHttpServerRequest(req))
-          .foldZIO(
-            error => ZIO.fail(error),
-            {
-              case RequestResult.Response(resp) =>
-                resp.body match {
-                  case None              => handleHttpResponse(resp, None)
-                  case Some(Right(body)) => handleHttpResponse(resp, Some(body))
-                  case Some(Left(body))  => handleWebSocketResponse(body)
-                }
-
-              case RequestResult.Failure(_) =>
-                ZIO.fail(
-                  new RuntimeException(
-                    s"The path: ${req.path} matches the shape of some endpoint, but none of the " +
-                      s"endpoints decoded the request successfully, and the decode failure handler didn't provide a " +
-                      s"response. ZIO Http requires that if the path shape matches some endpoints, the request " +
-                      s"should be handled by tapir."
-                  )
-                )
-            }
-          )
-      }
+      interpreter
+        .apply(ZioHttpServerRequest(req))
+        .foldZIO(
+          error => ZIO.fail(Response.internalServerError(error.getMessage)),
+          {
+            case RequestResult.Response(resp) =>
+              resp.body match {
+                case None              => handleHttpResponse(resp, None)
+                case Some(Right(body)) => handleHttpResponse(resp, Some(body))
+                case Some(Left(body))  => handleWebSocketResponse(body)
+              }
+
+            case RequestResult.Failure(_) =>
+              ZIO.fail(Response.internalServerError(
+                s"The path: ${req.path} matches the shape of some endpoint, but none of the " +
+                  s"endpoints decoded the request successfully, and the decode failure handler didn't provide a " +
+                  s"response. ZIO Http requires that if the path shape matches some endpoints, the request " +
+                  s"should be handled by tapir."
+              ))
```

Comment:
I am still trying to figure out this part. I think exposing so many details about a server with the response body is not a good idea. Would it be better to use `ZIO.logError` with a detailed message and response with a limited one?
```scala
ZIO.logError(
  s"The path: ${req.path} matches the shape of some endpoint, but none of the " +
    s"endpoints decoded the request successfully, and the decode failure handler didn't provide a " +
    s"response. ZIO Http requires that if the path shape matches some endpoints, the request " +
    s"should be handled by tapir."
) *> ZIO.fail(
  Response.internalServerError(
    s"The path: ${req.path} matches the shape of some endpoint, but none of the " +
      s"endpoints decoded the request successfully, and the decode failure handler didn't provide a " +
      s"response.
  )
)
```